core/gas.c

```diff
@@ -139,3 +140,26 @@ void fill_pressures(struct gas_pressures *pressures, const double amb_pressure,
 		}
 	}
 }
+
+enum gastype gasmix_to_type(struct gasmix mix)
+{
+	if (gasmix_is_air(mix))
+		return GASTYPE_AIR;
+	if (mix.he.permille == 0)
+		return mix.o2.permille >= 230 ? GASTYPE_NITROX : GASTYPE_AIR;
+	return mix.he.permille <= 300 ? GASTYPE_NORMOXIC : GASTYPE_TRIMIX;
```

Comment:
So I am confused by that. 
The definition of **normoxic** that I am familiar with is a Trimix gas that has roughly normal oxygen content. Every agency thinks about the details differently, but a quick google search landed me here http://www.ddcbuildsite.com/index.php/easyblog/entry/normoxic-trimix-diving-diving-deep-with-a-clearer-head which suggests O2 values between 18-21% -- I was used to 19-22%
I have never heard of the term **normoxic** (which, etymologically seems to imply something like 'normal oxygen') to be triggered by the H2 percentage as in this formula...
If we want to go that route, then one other term that I hear thrown around a lot with trimix divers (full disclosure, I'm a TDI tech dive master and have dive-mastered (shouldn't say 'taught') advanced trimix classes in the past) is **hypoxic** (so hypo or below oxygen) for trimix gasses with an O2 content below whatever you pick for the lower threshold above - in my training 18/40 was considered **hypoxic**, but 19/40 was considered **normoxic**

I am NOT saying that these are the definitions we have to use - all I'm saying is that the definition above has me rather confused...

Reply:
I originally had hypoxic, but I believe that @willemferguson objected.
See https://github.com/subsurface/subsurface/pull/3052#issuecomment-731947533 and follow-up discussion.
Perhaps I misunderstood things. The good thing is that this is trivial to change by changing gas.h and gas.c.

Reply:
I can look through that discussion again and see how we got here - I'll focus on the rest, first, to optimize time :-)

Reply:
Ok, found it: https://github.com/subsurface/subsurface/pull/3052#issuecomment-732253281
> With respect to the hypoxic gas class, I would recommend that you merge it with trimix. Full trimix dives are characterised by a hypoxic oxygen concentration (< 18%) and Helium < 30%. In contrast, normoxic dives are usually defined by oxygen around 20% and helium <= 30%. I would say that in a grey area the critical distinguishing factor is the oxygen concentration.

So this is very much what I talked about above, just with `< 18` instead of `<= 18` as in my training.
But I think his underlying comment kinda was to drop hypoxic (and not to add normoxic) and just have Air, Nitrox, Trimix (and maybe Oxygen as extreme Nitrox).
That said, I think it would be interesting to list all the classes (we can always add 12 million configuration options later... on second thought, let's NOT add all those options).
Outside of training dives I don't think you'll find a lot of trimix with O2 above 22%, so we could call that "Trimix" and the other two "Normoxic Trimix" and "Hypoxic Trimix". Makes sense?

Reply:
> Outside of training dives I don't think you'll find a lot of trimix with O2 above 22%, so we could call that "Trimix" and the other two "Normoxic Trimix" and "Hypoxic Trimix". Makes sense?

Fine with me. Note that I think that @willemferguson's argument was that, precisely because hypoxic is the standard Trimix, mixtures <= 18% should be called "Trimix" without further qualifier. I will just qualify all three and you can then discuss which one is the "Trimix" version. FWIW, according to Wikipedia hyperoxic Trimix is also called "TriOx".